### PR TITLE
Fix(fluffy samplesheet)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

PR to re-route the flow to new illumina tables for the flufyf sample sheet creation

NOTE: A temporary patch for a test was introduced to make it passing. Will be corrected in a future PR as it also requires refactoring which is out of scope for this PR.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/crud/read.py [hunk 1]
```diff
@@ -408,55 +400,6 @@ def get_number_of_reads_for_sample_passing_q30_threshold(
...
-
-    def get_number_of_reads_for_flow_cell(self, flow_cell_name: str) -> int:
-        """Get total number of reads for a flow cell from sample lane sequencing metrics."""
-        sequencing_metrics: list[SampleLaneSequencingMetrics] = (
-            self.get_sample_lane_sequencing_metrics_by_flow_cell_name(flow_cell_name=flow_cell_name)
-        )
-        read_count: int = 0
-        for sequencing_metric in sequencing_metrics:
-            read_count += sequencing_metric.sample_total_reads_in_lane
-        return read_count
```
Comment (ChrOertlin) [inline cg/store/crud/read.py:458]: These were not used anymore due to the previous refactoring

File: cg/store/crud/read.py [hunk 2]
```diff
@@ -331,14 +331,6 @@ def get_latest_ticket_from_case(self, case_id: str) -> str:
         """Returns the ticket from the most recent sample in a case."""
         return self.get_case_by_internal_id(internal_id=case_id).latest_ticket
 
-    def get_latest_flow_cell_on_case(self, family_id: str) -> Flowcell:
-        """Fetch the latest sequenced flow cell related to a sample on a case."""
-        flow_cells_on_case: list[Flowcell] = self.get_flow_cells_by_case(
-            case=self.get_case_by_internal_id(internal_id=family_id)
-        )
-        flow_cells_on_case.sort(key=lambda flow_cell: flow_cell.sequenced_at)
-        return flow_cells_on_case[-1] if flow_cells_on_case else None
```
Comment (ChrOertlin) [inline cg/store/crud/read.py:340]: Is now not used anymore

File: tests/cli/workflow/fluffy/test_cli_start.py [hunk 3]
```diff
@@ -6,16 +6,20 @@
...
 
 def test_start_available_dry(
-    cli_runner: CliRunner, fluffy_case_id_existing: str, fluffy_context: CGConfig, caplog
+    cli_runner: CliRunner, fluffy_case_id_existing: str, fluffy_context: CGConfig, caplog, mocker
 ):
     caplog.set_level("INFO")
 
     # GIVEN a case_id that does exist in database
 
+    # Temporary patch to avoid error in the test with the devel branch will be addressed in the future
```
Comment (ChrOertlin) [inline tests/cli/workflow/fluffy/test_cli_start.py:20]: The mocked function will be refactored in a future pr. At that point this test should pass with the new functionality and this line will be removed

File: cg/meta/workflow/fluffy.py [hunk 4]
```diff
@@ -148,7 +150,7 @@ def link_fastq_files(self, case_id: str, dry_run: bool = False) -> None:
         for family_sample in case_obj.links:
             sample_id = family_sample.sample.internal_id
             files: Query = self.housekeeper_api.files(
-                bundle=sample_id, tags=["fastq", latest_flow_cell.name]
+                bundle=sample_id, tags=[SequencingFileTag.FASTQ, sequencing_run.device.internal_id]
```
Comment (Vince-janv) [inline cg/meta/workflow/fluffy.py:153]: ✨

File: tests/cli/workflow/fluffy/test_cli_start.py (hunk 3 above)
Comment (Vince-janv) [reply tests/cli/workflow/fluffy/test_cli_start.py:20]: I would add @pytest.mark.xfail(reason="flow cell on-disk check is yet to be refactored") to the test instead.

It basically marks the test as expected to fail and doesn't complain if it does 👍

**Review by Vince-janv — APPROVED** [review]
So nice with PRs that remove more code than they add 🌟

File: tests/cli/workflow/fluffy/test_cli_start.py (hunk 3 above)
Comment (ChrOertlin) [reply tests/cli/workflow/fluffy/test_cli_start.py]: added

**Pull request merged**